assets/prototype/application/ui/components/layout/editor-modal/useEditorModalState.ts

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { useReducer, useEffect } from 'react';
+import { drop, head } from 'ramda';
+
+const CLOSE_ALL = 'CLOSE_ALL';
+const CLOSE_EDITOR = 'CLOSE_EDITOR';
+const OPEN_EDITOR = 'OPEN_EDITOR';
+
+type ActionType = 'CLOSE_ALL' | 'CLOSE_EDITOR' | 'OPEN_EDITOR';
+
+type State = string[];
+
+type Action = {
+	editorId?: string;
+	type: ActionType;
+};
+
+const reducer = (state: State, action: Action) => {
+	const { editorId } = action;
+
+	switch (action.type) {
+		case CLOSE_EDITOR:
+			if (editorId === head(state)) {
+				const stateWithoutFirstItem = drop(1, state);
+				return stateWithoutFirstItem;
+			}
+			return state;
+		case CLOSE_ALL:
+			return [];
+		case OPEN_EDITOR:
+			if (editorId && !state.includes(editorId)) {
+				return [...state, editorId];
+			}
+			return state;
+		default:
+			throw new Error('Unexpected action');
+	}
+};
+
+const useEditorModalState = () => {
+	const initialState = [];
+	const [state, dispatch] = useReducer(reducer, initialState);
+
+	useEffect(() => {
+		console.log('useEditorModalState >>>', state);
+	}, [state]);
+
+	const openEditor = (editorId) => {
+		dispatch({
+			type: OPEN_EDITOR,
+			editorId,
+		});
+	};
+
+	const closeEditor = (editorId) => {
```

Comment:
Better to make it `editorId: string`

Reply:
Added: https://github.com/eventespresso/event-espresso-core/pull/2074/commits/481f0310ad52b9bad886b498d4073c545af54e8d
Thx :bowing_man: